packages/api/src/share_link/controller.test.ts

```diff
@@ -611,29 +608,29 @@ describe("GET /share-links", () => {
 	});
 
 	test("should return 400 if the header values are invalid", async () => {
-		(verifyUserAuthentication as jest.Mock).mockImplementation(
-			(_: Request, __: Response, next: NextFunction) => {
+		(extractUserEmailFromAuthToken as jest.Mock).mockImplementation(
+			(req: Request, __: Response, next: NextFunction) => {
+				(
+					req.body as {
+						emailFromAuthToken: string;
+						userSubjectFromAuthToken: string;
+					}
+				).emailFromAuthToken = "not_an_email";
 				next();
 			},
 		);
 		await agent.get("/api/v2/share-links?shareLinkIds[]=1").expect(400);
 	});
 
 	test("should return an empty list if caller asks for share links they can't access", async () => {
-		(verifyUserAuthentication as jest.Mock).mockImplementation(
+		(extractUserEmailFromAuthToken as jest.Mock).mockImplementation(
```

Comment:
I hadn't seen this pattern (type casting a mocked function as mocked) before so looked into it.  The motivation is just that as you know type casts are an override to the static analysis / if we're wrong about the cast then it causes problems.

It looks like the safer alternative to this approach would be to use [`jest.mocked()`](https://jestjs.io/docs/mock-function-api#jestmockedsource-options).

So you could replace this line with:

```
jest.mocked(extractUserEmailFromAuthToken).mockImplementation(...)
```

And you'd get the mocking type without the type cast!